### PR TITLE
Backup Client Resource

### PR DESCRIPTION
**Pull request by dotIN**
- New Backup Install Client Resource added
- Unit tests added for the same
- InSepc tests added as well
- Changes made in .kitchen to pick new tests

### CODE REVIEW

File: resources/install_client.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
+#
+# Cookbook:: bitbucket_backup_client
+# Resource:: install Backup Client
+#
+resource_name :install_backup_client
```
Comment (bharathcp) [inline resources/install_client.rb]: name it just backup_client. verb is the action and not the resource

File: resources/install_client.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+property :product, String, default: 'bitbucket'
+property :version, String, default: '3.3.2'
+property :bitbucket_user, String, default: 'atlbitbucket'
+property :bitbucket_group, String, default: 'atlbitbucket'
+property :home_path, String, default: '/var/atlassian/application-data/bitbucket'
+property :install_path, String, default: '/opt/atlassian'
+property :client_url, String, default: 'https://maven.atlassian.com/content/groups/public/com/atlassian/bitbucket/server/backup/bitbucket-backup-distribution'
+property :bitbucket_url, String, default: 'http://127.0.0.1:8080'
+property :backup_path, String
```
Comment (bharathcp) [inline resources/install_client.rb]: required=true?

File: templates/backup-config.properties.erb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+# Defines the location of the home directory of the Bitbucket Server instance you wish to back up or restore to. REQUIRED
+# If omitted here it will be taken from the BITBUCKET_HOME environment variable or the Java system property of the same name if supplied to the Backup and Restore Client on the command line. As a required value, backup and restore will fail if it is not supplied through one of these mechanisms.
+bitbucket.home=<%= @backup_client['home_path'] %>
+# Defines the username of the Bitbucket Server user with administrative privileges you wish to perform the backup. REQUIRED
+# If omitted here it will be taken from the Java system property of the same name if supplied to the Backup Client on the command line. 
+bitbucket.user=<%= @backup_client['user'] %>
+# Defines the password of the Bitbucket Server user with administrative privileges you wish to perform the backup. REQUIRED
+# If omitted here it will be taken from the Java system property of the same name if supplied to the Backup Client on the command line.
+bitbucket.password=<%= @backup_client['password'] %>
```
Comment (bharathcp) [inline templates/backup-config.properties.erb:11]: test if password is empty

File: test/cookbooks/test/recipes/backup_client.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+#
+# Cookbook:: bitbucket_server
+# Recipe:: Install Backup Client
+#
+# Copyright:: 2017, The Authors, All Rights Reserved.
+package 'unzip'
```
Comment (bharathcp) [inline test/cookbooks/test/recipes/backup_client.rb:6]: call out this expectation in readme or add this to custom resource

File: test/smoke/default/backup_client_test.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
+# # encoding: utf-8
+
+# Inspec test for recipe bitbucket_server::default
+
+# The Inspec reference, with examples and extensive documentation, can be
+# found at http://inspec.io/docs/reference/resources/
+
+describe file('/var/atlassian/application-data/bitbucket/shared/backup-config.properties') do
+  it { should exist }
```
Comment (bharathcp) [inline test/smoke/default/backup_client_test.rb:9]: need to test the ownership and other attributes here

File: test/smoke/default/backup_client_test.rb [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+end
+
+
+describe file('/opt/atlassian/bitbucket-backup-client/bitbucket-backup-client.jar') do
+  it { should exist }
```
Comment (bharathcp) [inline test/smoke/default/backup_client_test.rb]: need to test the ownership and other attributes here

File: test/smoke/default/backup_client_test.rb [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+end
+
+
+describe file('/opt/atlassian/bitbucket-backup-client/bitbucket-restore-client.jar') do
+  it { should exist }
```
Comment (bharathcp) [inline test/smoke/default/backup_client_test.rb]: need to test the ownership and other attributes here

**Review by bharathcp — COMMENTED** [review]
add read me for backup client

File: resources/install_client.rb (hunk 1 above)
Comment (dotIN) [reply resources/install_client.rb]: Agree, also renaming the source to backup_client.rb

File: resources/install_client.rb (hunk 2 above)
Comment (dotIN) [reply resources/install_client.rb]: Done

File: templates/backup-config.properties.erb (hunk 3 above)
Comment (dotIN) [reply templates/backup-config.properties.erb:11]: Tested,empty is also fine

**Pull request merged**